# GenGrab
The is a bioinformatic tool to extract required gene segments from the entire genome according to defined gene boundary  
## Description
* The data results of gene boundaries from various softwares like ORF finder, Mitos can be executed in user defined format to extract required genes
* Users can get gene sequences from all kinds of genomes like linear/cercular and postive/negative stranded condition
* Results will be stored in user defined path in fasta format
## Getting started
### Installation
* The simplest way to set this up is to install the Anaconda Python distribution
* Otherwise, install the following dependencies: 1. pandas
                                                 2. tkinter
                                                 3. numpy
### Running the program
* Unzip the program folder
* Open Anaconda prompt
* Type “python“ followed by a space
* Drag and drop the “gengrab.py” file (which is in the folder you just unzipped) onto the command prompt.
           
           `python “<gengrab.py>”`

* Hit enter
* Give the genome sequence (test_genome.fas) from the dialog box just opened
* Give the gene boundaries in the required format(test_boundary.csv) to run the program
* Save the output file in required path from the dialog box
## Feedback to contributors
* **Goirik Dasgupta** @goirikdg
* **Avas Pakrashi** @avvy_paks
## Acknowledgments
* Code used : https://codereview.stackexchange.com/questions/151329/reverse-complement-of-a-dna-string/151333
* Thankful to Dr. Kailash Chandra, Director of Zoological Survey of India.




